modules/effects/src/provide_effects.ts

```diff
@@ -44,10 +44,11 @@ import { rootEffectsInit as effectsInit } from './effects_actions';
  * ```
  */
 export function provideEffects(
-  ...effects: Type<unknown>[]
+  ...effects: Type<unknown>[] | Type<unknown>[][]
 ): EnvironmentProviders {
```

Comment:
I'd add two signatures here to avoid the ability to pass multiple arrays, for example:

```ts
provideEffects([Effects1, Effects2], [], [Effects3]); // this would work with current impl
```

```suggestion
export function provideEffects(effects: Type<unknown>[]): EnvironmentProviders;
export function provideEffects(...effects: Type<unknown>[]): EnvironmentProviders;
export function provideEffects(
  ...effects: Type<unknown>[] | Type<unknown>[][]
): EnvironmentProviders {
```

Reply:
I started out like that, but then we lose the TS Docs - or it should be duplicated for all signatures. Or is there a way around that?

Reply:
I think if we put the TSDoc comment above the implementation, API docs will be generated properly:

```ts
export function provideEffects(effects: Type<unknown>[]): EnvironmentProviders;
export function provideEffects(...effects: Type<unknown>[]): EnvironmentProviders;
/**
 * ...
 */
export function provideEffects(
  ...effects: Type<unknown>[] | Type<unknown>[][]
): EnvironmentProviders {
  // ...
}
```

Reply:
Not that I can see:

![image](https://user-images.githubusercontent.com/28659384/199717828-0317dc20-121e-491b-93f1-cb28044fffff.png)

While adding it to the correct overload results in:

![image](https://user-images.githubusercontent.com/28659384/199717899-12274ff4-f8e4-4e95-af40-f8cccd1ed0e8.png)

Reply:
We can add descriptions for signatures too, but the `@usageNotes` section needs to be above the implementation, otherwise it won't be displayed in the [API docs](http://localhost:4200/api/effects/provideEffects). I've just checked this locally.

```ts
/**
 * description for signature 1
 */
export function provideEffects(effects: Type<unknown>[]): EnvironmentProviders;
/**
 * description for signature 2
 */
export function provideEffects(...effects: Type<unknown>[]): EnvironmentProviders;
/**
 * description
 *
 * @usageNotes
 *
 * ### Providing effects at the root level
 *
 * ...
 */
export function provideEffects(
  ...effects: Type<unknown>[] | Type<unknown>[][]
): EnvironmentProviders {
  // ...
}
```

Reply:
Oh cool, I didn't know that.
Let me verify it later today, thanks @markostanimirovic !